### ISSUE TITLE
Suspense not working in a nested route 

### ISSUE THREAD

**Issue by mbarekTrismegistus** [vercel/next.js#67736]
### Link to the code that reproduces this issue

https://github.com/HermesThemes/ArtDungeon

### To Reproduce

1-npm run dev
2- navigate to /post/[id]

### Current vs. Expected behavior

after step 2 you can see that is show the data directly and if i added a sleep function of 5 second it will wait all that time and not showing a suspense fallback
I expect to the the suspense fallback just like in other route (note: suspense is not working only on a nested route, folder inside folder)

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Wed, 17 Apr 2024 15:20:28 +0000
  Available memory (MB): 7845
  Available CPU cores: 4
Binaries:
  Node: 21.7.3
  npm: 10.5.2
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.4
  eslint-config-next: 14.2.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure, Partial Prerendering (PPR)

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

This bug seems to happen only in a nested route, as the same component or page will work very well in the main route, I tested the same component (Post) in in a dynamic route and it's not working but works on the main page.js 

https://github.com/user-attachments/assets/6690a23b-f932-4596-80a3-0a02c07bc511

**Comment by flacial** [comment]
Same here. Found a fix @mbarekTrismegistus ?

**Comment by mbarekTrismegistus** [comment]
> Same here. Found a fix @mbarekTrismegistus ?

yes the only solution is renaming that folder, ik it's weird but naming it fixed the issues , that's soo weird

**Comment by marcoshernanz** [comment]
Thanks for the report. I could not reproduce this with the linked project. On `next@14.2.4` and Node 21.7.3, adding a five-second delay to the async child under `/post/[id]` streams and displays `loading.b..` before the child resolves. Current stable and canary releases behave the same.

The committed reproduction does not include that delay in the nested child, and the later fix changes both the route and data-fetching structure, so it does not isolate a route-naming bug. I’m closing this as not reproducible. If this still occurs, please open a new issue with a minimal repository, the browser and version, and identical navigation steps for both routes.
